components/button/src/index.js

```diff
@@ -38,8 +39,15 @@ const StyledButton = styled('button')(
     textDecoration: 'none',
     WebkitAppearance: 'none',
     WebkitFontSmoothing: 'antialiased',
+    ':hover': {
```

Comment:
looking around a bit wider at the rest of the styling for button here, i'm not sure how it matches up with GDS

the example here https://govuk-elements.herokuapp.com/buttons/ seems to use a slightly different approach especially WRT how `top` works.  the `:focus` state for example doesn't adjust `top`, however `:active` does